### ISSUE TITLE
Ongoing Limitations Issue

### ISSUE THREAD

**Issue by willtebbutt** [withbayes/Taped.jl#31]
This issue is going to remain permanently open. It should not ever be closed. Its purpose is to track known test failure cases, and their causes.

### Distributions.jl

1. `Rmath.dnbeta`, `Rmath.dnchisq`, `Rmath.dnf`, and `Rmath.dnt` all use `ccall`s for which it no one has derived an `rrule` -- this is presumably because they involve infinite series, and it is unclear how to write down a rule for them.
1. `logpdf` for `Chernoff` takes a really long time to run. It involves quadrature, so presumably has quite a substantial inner-loop, which is probably why it takes so long.
1. `logpdf` for`NormalInverseGamma` involves the [`_besselk`](https://github.com/JuliaMath/SpecialFunctions.jl/blob/be1fa06fee58ec019a28fb0cd2b847ca83a5af9a/src/bessel.jl#L261C10-L261C18) function, which contains a `ccall` for which we have no `rrule`. Unfortunately, the [`rrule` for `besselk`](https://github.com/JuliaMath/SpecialFunctions.jl/blob/be1fa06fee58ec019a28fb0cd2b847ca83a5af9a/ext/SpecialFunctionsChainRulesCoreExt.jl#L104) returns a `ChainRulesCore.NotImplemented` w.r.t. its first argument. This means that we cannot simply wrap this `rrule`, because we would immediately `increment!!` the tangent, which will cause an error to be thrown by ChainRulesCore. In order to support this, we need to modify this framework to permit inactive arguments to `rrule!!`s, and to take advantage of this in practice.
1. A `Dirichlet` test does not pass due to control-flow changing between runs. I'm not entirely sure why this is happening at present, but it ought to require only a tweak to testing to fix.

**Comment by acertain** [comment]
Seems like Tapir doesn't support try/catch (I got a `ERROR: Tapir.UnhandledLanguageFeatureException("Encountered UpsilonNode: ϒ (%153)")`), maybe this should be mentioned in the docs?

**Comment by willtebbutt** [comment]
Yeah, this should probably be mentioned. Are you able to provide a MWE for this error? I'm aware that we don't handle `Upsilon` nodes, but I've not encountered them in many of the programmes that we've tried to differentiate.

**Comment by willtebbutt** [comment]
My point being that, while we don't have _complete_ support for all `try` / `catch` blocks, it's not that you can't use them at all.

**Comment by acertain** [comment]
MWE: `@warn` with string interpolation:

```
using Tapir, DifferentiationInterface

function f(x)
  @warn "$x"
  return x*x
end

DifferentiationInterface.gradient(f, AutoTapir(), 2)
```
